src/store/conversationsStore.js

```diff
@@ -203,14 +203,42 @@ const mutations = {
 
 const actions = {
 	/**
-	 * Add a conversation to the store and index the displayName.
+	 * Add a conversation to the store and index the displayName
 	 *
 	 * @param {object} context default store context;
 	 * @param {object} conversation the conversation;
 	 */
 	addConversation(context, conversation) {
 		context.commit('addConversation', conversation)
 
+		context.dispatch('postAddConversation', conversation)
+	},
+
+	/**
+	 * Add conversation to the store only, if it was changed according to lastActivity and modifiedSince
+	 *
+	 * @param {object} context dispatch context
+	 * @param {object} payload mutation payload
+	 * @param {object} payload.conversation the conversation
+	 * @param {number|0} payload.modifiedSince timestamp of last state or 0 if unknown
+	 */
+	addConversationIfChanged(context, { conversation, modifiedSince }) {
+		if (conversation.lastActivity >= modifiedSince) {
+			context.commit('addConversation', conversation)
+		}
```

Comment:
```suggestion
		} else {
		  // Smart update
		}
```

Don't we want to apply smartUpdate here to cover rest cases (full re-fetch at least)?

Reply:
If we use the smartUpdate approach in else, then we do not need `if` branch, as `smartUpdate` covers all possible updates anyway